src/base/net/geoipmanager.cpp

```diff
@@ -420,7 +433,16 @@ void GeoIPManager::downloadFinished(const DownloadResult &result)
     }
 
     QString error;
-    GeoIPDatabase *geoIPDatabase = GeoIPDatabase::load(data, error);
+    const QHash<QString, QByteArray> files = Utils::Tar::untar(data, error);
+
+    GeoIPDatabase *geoIPDatabase = nullptr;
+    for (auto iter = files.cbegin(); iter != files.cend(); ++iter) {
+        if (!iter.key().endsWith(GEOIP_FILENAME))
+            continue;
+
+        geoIPDatabase = GeoIPDatabase::load(iter.value(), error);
+    }
+
     if (geoIPDatabase) {
         if (!m_geoIPDatabase || (geoIPDatabase->buildEpoch() > m_geoIPDatabase->buildEpoch())) {
             if (m_geoIPDatabase)
```

Comment:
Slightly unrelated to this PR but there are instances in geoipmanager where you can omit the null check here, [delete in c++ is a noop on null pointers](https://isocpp.org/wiki/faq/freestore-mgmt#delete-handles-null). If you want to keep that seperate I'm more than happy to address that in a seperate PR.

Reply:
Addressed in #11961